crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,1418 @@
+///     QkPropsMap *props_map = qk_property_map_new();
+///     qk_property_map_add(props_map, NULL, 0, qk_instruction_properties_new(0.0, 0.1));
+///     qk_target_add_instruction(target, QkGate_CRX, *[3.14], props_map);
+///     qk_target_add_instruction(target, QkGate_H, NULL, qk_property_map_new());
+///
+///     qk_target_operation_names(target)
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_operation_names(target: *const Target) -> *mut *mut c_char {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    let mut names: Vec<*mut c_char> = target
+        .operation_names()
+        .map(|name| CString::new(name).unwrap().into_raw())
+        .collect();
+    let pointer = names.as_mut_ptr();
+
+    // Prevent vec from being destroyed
+    forget(names);
+    pointer
+}
+
+/// @ingroup QkTarget
+/// Retrieves the physical qubits in this Target.
+///
+/// @param target A pointer to the Target.
+///
+/// @return An array with all of the physical qubits in the Target.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     qk_target_phyisical_qubits(target)
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_phyisical_qubits(target: *const Target) -> *mut usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    let mut physical_qubits: Vec<usize> = target.physical_qubits().collect();
+
+    physical_qubits.as_mut_ptr()
+}
+
+/// @ingroup QkTarget
+/// Retrieves the names of all non-global operations in this Target.
+///
+/// @param target A pointer to the Target.
+/// @param strict_direction Checks whether the direction of the instruction's
+/// qargs should be considered when classifying it.
+///
+/// @return A list of the names for the global operations in the Target.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     QkPropsMap *props_map = qk_property_map_new();
+///     qk_property_map_add(props_map, NULL, 0, qk_instruction_properties_new(0.0, 0.1));
+///     qk_target_add_instruction(target, QkGate_CRX, *[3.14], props_map);
+///     qk_target_add_instruction(target, QkGate_H, NULL, qk_property_map_new());
+///
+///     qk_target_non_global_operation_names(target, true)
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_non_global_operation_names(
+    target: *mut Target,
+    strict_direction: bool,
+) -> *mut *mut c_char {
```

Comment:
This looks very strange, how do we know the length of the outer vector? Would it make more sense to have a dedicated struct that users can query from? 

Another approach could also be to not use `*char` to describe the operations, since in C we only have the standard gates anyways. Maybe we can also discuss this offline.

Reply:
A struct would make more sense here since we can store the length of the output. I will make sure to add that.